### PR TITLE
Feature/add reviews and calculated rating

### PR DESCRIPTION
**Pull request by zatkiller**
Updated Reviews Constraint, use JWT user id, updated docs and implemented error handling

### CODE REVIEW

File: src/controllers/reviewController.ts [hunk 1]
```diff
@@ -48,17 +51,27 @@ async function showReview(req: Request, res: Response, next: NextFunction) {
 
 async function createReview(req: Request, res: Response, next: NextFunction) {
   try {
+    const token = req.header('x-auth-token');
```
Comment (ivantjh) [inline src/controllers/reviewController.ts]: The decoding is already done in the `auth` middleware. Just need to do
```typescript
try {
    const review = await Review.create({ ...req.body, userId: req.userId, stallId: req.params.Id })
    res.status(201).json(review);
} catch(err) {
    ....
}
```

File: src/controllers/reviewController.ts [hunk 2]
```diff
@@ -48,17 +51,27 @@ async function showReview(req: Request, res: Response, next: NextFunction) {
...
+    if (err instanceof UniqueConstraintError)
+      next(new BadRequestError('A review for this stall already exists!'));
+
     next(err);
   }
 }
 
 async function updateReview(req: Request, res: Response, next: NextFunction) {
   try {
+    const token = req.header('x-auth-token');
```
Comment (ivantjh) [inline src/controllers/reviewController.ts]: `userId` is already in req -> no need decode again

File: src/middleware/class/class.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export class JWTPayload {
```
Comment (ivantjh) [inline src/middleware/class/class.ts]: Can remove this, the exact same interface is available as `UserDecoded` in `auth.ts`

File: package.json [hunk 4]
```diff
@@ -15,6 +15,7 @@
     "express": "^4.17.1",
     "fast-csv": "^4.3.6",
     "jsonwebtoken": "^8.5.1",
+    "jwt-decode": "^3.1.2",
```
Comment (ivantjh) [inline package.json]: No need additional library, `jwt.verify` already provides the decode. Can remove

**Review by ivantjh — CHANGES_REQUESTED** [review]

**Review by wenhaogoh — APPROVED** [review]
LGTM

**Pull request merged**